conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -182,12 +184,26 @@ def _get_cmake_lib(self, info, components, pkg_folder, pkg_folder_var):
                   "cflags": " ".join(info.cflags),
                   "sharedlinkflags": " ".join(info.sharedlinkflags),
                   "exelinkflags": " ".join(info.exelinkflags),
-                  "system_libs": system_libs}
-
-        if info.frameworks:
-            ConanOutput(scope=str(self._conanfile)).warning("frameworks not supported yet in new CMakeDeps generator")
-
-        if info.libs:  # TODO: to change to location
+                  "system_libs": system_libs
+        }
+        if is_apple_os(self._root_conanfile):
```

Comment:
Isn't the responsibility of the package to define ``framework`` only when appropiate? In the same way you can't define ``system_libs = ["win32_sock"]`` if you are not in Windows.